pkg/cmd/config/config_keys.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/spf13/cobra"
+)
+
+// newCmdConfigKeys returns the keys that a user can set in a configuration file.
+func newCmdConfigKeys(ctx api.Context) *cobra.Command {
+	var quietOutput bool
+	cmd := &cobra.Command{
+		Use:   "keys",
+		Short: "Print all the keys that can be set in a configuration file",
+		Args:  cobra.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			keys := []string{
+				"core.dcos_acs_token",
+				"core.dcos_url",
+				"core.mesos_master_url",
+				"core.pagination",
+				"core.ssl_verify",
+				"core.timeout",
+				"core.ssh_user",
+				"core.ssh_proxy_ip",
+				"core.reporting",
+				"core.prompt_login",
+				"cluster.name",
+				"job.url",
+				"job.service_name",
+				"marathon.url",
+				"package.cosmos_url",
+			}
+
+			configMap := map[string]string{
+				"core.dcos_acs_token":   "the DC/OS authentication token",
+				"core.dcos_url":         "the public master URL of your DC/OS cluster",
+				"core.mesos_master_url": "the Mesos master URL (defaults to 'core.dcos_url')",
+				"core.pagination":       "indicates whether to paginate output (defaults to true)",
+				"core.ssl_verify":       "indicates whether to verify SSL certificates or set the path to the SSL certificates",
+				"core.timeout":          "the request timeout in seconds, with a minimum value of 1 second (defaults to 3 minutes)",
+				"core.ssh_user":         "the user used when using ssh to connect to a node of your DC/OS cluster (defaults to 'core')",
+				"core.ssh_proxy_ip":     "whether to use a fixed ssh proxy host (Bastion) for node SSH access",
+				"core.reporting":        "whether to report usage events to Mesosphere",
+				"core.prompt_login":     "whether to prompt the user to log in when token expired, otherwise automatically initiate login",
+				"cluster.name":          "human readable name of cluster",
+				"job.url":               "API URL for talking to the Metronome scheduler",
+				"job.service_name":      "the name of the metronome cluster",
+				"marathon.url":          "base URL for talking to Marathon, overwrites the value specified in 'core.dcos_url'",
+				"package.cosmos_url":    "base URL for talking to Cosmos, overwrites the value specified in 'core.dcos_url'",
+			}
+			for _, key := range keys {
```

Comment:
you can loop through the map, that removes the need for the slice

Reply:
Ordering is not offered when looping over a map, that's why I prefer to loop over a slice so that the keys are alphabetically organized. I could order the keys at runtime to give the same output.

Reply:
Ah I see, yeah let's order at runtime from the map, that'd make it less likely that we forget a place when adding new keys. Also if it's returned from the config package it's easier to keep the list in sync.